sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/CommunicationIdentityClientTestBase.java

```diff
@@ -30,6 +36,25 @@ protected CommunicationIdentityClientBuilder getCommunicationIdentityClient(Http
         builder.endpoint(ENDPOINT)
             .accessKey(ACCESSKEY)
             .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient);
+       
+        if (getTestMode() == TestMode.RECORD) {
+            builder.addPolicy(interceptorManager.getRecordPolicy());
+        }
+
+        return builder;
+    }
+
+    protected CommunicationIdentityClientBuilder getCommunicationIdentityClientBuilderUsingManagedIdentity(HttpClient httpClient) {
+        CommunicationIdentityClientBuilder builder = new CommunicationIdentityClientBuilder();
+        builder
+            .endpoint(ENDPOINT)
+            .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient);
+        
+        if (getTestMode() == TestMode.PLAYBACK) {
+            builder.credential(new FakeCredentials());
```

Comment:
Instead of implementing a `FakeCredential` you can just use the `AzureKeyCredential` and delete your class.

```suggestion
            builder.credential(new AzureKeyCredential("invalid key"));
```

Reply:
But AzureKeyCredential doesn't implement TokenCredential interface